tests/epics/demo/test_demo.py

```diff
@@ -99,11 +99,11 @@ async def test_mover_moving_well(sim_mover: demo.Mover) -> None:
 
 async def test_mover_stopped(sim_mover: demo.Mover):
     callbacks = []
-    set_sim_callback(sim_mover.stop_, lambda r, v: callbacks.append(v))
 
-    assert callbacks == [None]
-    await sim_mover.stop()
-    assert callbacks == [None, None]
+    with set_sim_callback(sim_mover.stop_, lambda r, v: callbacks.append(v)):
+        assert callbacks == [None]
+        await sim_mover.stop()
+        assert callbacks == [None, None]
```

Comment:
I wonder if `set_sim_callbacks` is what we actually want here? We add a callback to a signal, effectively monitoring it, but we use it to track when it's put to. Maybe we instead want:

```python
with track_sim_puts(sim_mover.stop_) as puts:
    assert puts == []
    await sim_mover.stop()
    assert puts == [None]
```

@rosesyrett what do you reckon?

Reply:
I just changed it to work, have no opinion on whether the original was correct. Though I think `set_sim_callbacks` is still useful and if we are going to change this test we should add another test for `set_sim_callbacks`